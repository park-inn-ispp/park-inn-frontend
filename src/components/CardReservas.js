import React from "react";
import PropTypes from "prop-types";
import call from "../Util/Caller";
import Leyenda from "./Leyenda";
import "./card.css";
function cancelarReserva(reservaId) {

  call(`/reservas/`+reservaId+'/cancelar', 'GET')
    .then(response => {
      console.log(response)

      if (response.ok){
        //navigate(`/mis-reservas`)
        window.location.reload()
      }
    })
}


function CardReservas({ imageSource,id, title,propietario,fechaInicio,fechaFin,precioTotal, estado, urlCancelar,urlDetalles }) {
  let canceledDiv =  <div className="cancelada"></div>
  let pendentDiv =  <div className="pendiente"></div>
  let aceptedDiv =  <div className="aceptada"></div>

  return (
    <div className="card text-center bg-dark animate__animated animate__fadeInUp">

      <div className="card-body text-light">
        <h4 className="card-title">{title}</h4>
        <div className="card-text-div">
        <p className="card-text text-secondary">
          {(estado=='cancelada' || estado=='rechazada')
            ? canceledDiv
            : (estado=='pendiente')
              ? pendentDiv
              : aceptedDiv
              }
        </p>
        <p className="card-text text-secondary">
          {propietario
            ? "Propietario : " + propietario 
            : "Propietario : -"}
        </p>
        <p className="card-text text-secondary">
          {fechaInicio
            ? "Fecha Inicio : " + fechaInicio 
            : "Fecha Inicio : -"}
        </p>
        <p className="card-text text-secondary">
          {fechaFin
            ? "Fecha Fin : " + fechaFin 
            : "Fecha Fin : -"}
        </p>
        <p className="card-text text-secondary">
          {precioTotal
            ? "Precio total : " + precioTotal +"â‚¬"
            : "Precio total : -"}
        </p>
        
        <a
          onClick={() => cancelarReserva(id)}
          className="btn btn-outline-secondary border-0"
          rel="noreferrer"
        >
         Cancelar

        </a>
        <a
          href={urlDetalles ? urlDetalles : "#!"}
          className="btn btn-outline-secondary border-0"
          rel="noreferrer"
        >
          Detalles
        </a>
        </div>
      </div>
    </div>
  );
}


export default CardReservas;