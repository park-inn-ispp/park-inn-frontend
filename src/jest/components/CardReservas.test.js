import CardReservas from '../../components/CardReservas'; 
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import navegarSnapshot from "../testUtility/navegarSnapshots"
Enzyme.configure({adapter: new Adapter()});

test('Renderizar una CardReservas vacía', () => {

    const emptyCard = shallow(<CardReservas />);

    expect(emptyCard.text()).toEqual('Fecha Inicio : -Fecha Fin : -Precio total : -CancelarDetalles');
  
});

test('Renderizar una CardReservas no vacía', () => {

    const notEmptyCard = shallow(<CardReservas   id={"1"} title={"test 1 title"} propietario={"test 1 propietario"} 
    fechaInicio={"test 1 fechaInicio"} fechaFin={"test 1 fechaFin"} precioTotal={"test 1 precioTotal"}
    urlCancelar={"test 1 urlCancelar"} urlDetalles={"test 1 urlDetalles"}  />);

    expect(notEmptyCard.text()).toEqual("test 1 titleFecha Inicio : test 1 fechaInicioFecha Fin : test 1 fechaFinPrecio total : test 1 precioTotal€CancelarDetalles");
  
});

test('Renderizar una CardReservas sin titulo vacía', () => {

    const notEmptyCard = shallow(<CardReservas   id={"1"} propietario={"test 1 propietario"} 
    fechaInicio={"test 1 fechaInicio"} fechaFin={"test 1 fechaFin"} precioTotal={"test 1 precioTotal"}
    urlCancelar={"test 1 urlCancelar"} urlDetalles={"test 1 urlDetalles"}  />);

    expect(notEmptyCard.text()).toEqual("Fecha Inicio : test 1 fechaInicioFecha Fin : test 1 fechaFinPrecio total : test 1 precioTotal€CancelarDetalles");
  
});

test('Renderizar una CardReservas no vacía con estado cancelada', () => {

    const estado = "cancelada"

    const notEmptyCard = <CardReservas   id={"1"} propietario={"test 1 propietario"} 
    fechaInicio={"test 1 fechaInicio"} fechaFin={"test 1 fechaFin"} precioTotal={"test 1 precioTotal"}
     estado={estado} urlCancelar={"test 1 urlCancelar"} urlDetalles={"test 1 urlDetalles"}  />;

     const tree = renderer
     .create(notEmptyCard)
     .toJSON();

     var elemento = navegarSnapshot(tree,"div>div>p>div")

     expect(elemento.props["className"]).toBe("cancelada");
});

test('Renderizar una CardReservas no vacía con los diferentes estados', () => {

    const estado = "pendiente"

    const notEmptyCard = <CardReservas   id={"1"} propietario={"test 1 propietario"} 
    fechaInicio={"test 1 fechaInicio"} fechaFin={"test 1 fechaFin"} precioTotal={"test 1 precioTotal"}
     estado={estado} urlCancelar={"test 1 urlCancelar"} urlDetalles={"test 1 urlDetalles"}  />;

     const tree = renderer
     .create(notEmptyCard)
     .toJSON();

     var elemento = navegarSnapshot(tree,"div>div>p>div")
     
     expect(elemento.props["className"]).toBe(estado);
});

test('Renderizar una CardReservas no vacía con estado aceptada', () => {

    const estado = "aceptada"

    const notEmptyCard = <CardReservas   id={"1"} propietario={"test 1 propietario"} 
    fechaInicio={"test 1 fechaInicio"} fechaFin={"test 1 fechaFin"} precioTotal={"test 1 precioTotal"}
     estado={estado} urlCancelar={"test 1 urlCancelar"} urlDetalles={"test 1 urlDetalles"}  />;

     const tree = renderer
     .create(notEmptyCard)
     .toJSON();

     var elemento = navegarSnapshot(tree,"div>div>p>div")
     
     expect(elemento.props["className"]).toBe(estado);
});