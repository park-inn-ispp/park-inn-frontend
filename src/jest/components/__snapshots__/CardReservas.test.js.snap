// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderizar una CardReservas no vacía con estado cancelada 1`] = `
<div
  className="card text-center bg-dark animate__animated animate__fadeInUp"
>
  <div
    className="card-body text-light"
  >
    <h4
      className="card-title"
    />
    <div
      className="card-text-div"
    >
      <p
        className="card-text text-secondary"
      >
        <div
          className="cancelada"
        />
      </p>
      <p
        className="card-text text-secondary"
      >
        Propietario : test 1 propietario
      </p>
      <p
        className="card-text text-secondary"
      >
        Fecha Inicio : test 1 fechaInicio
      </p>
      <p
        className="card-text text-secondary"
      >
        Fecha Fin : test 1 fechaFin
      </p>
      <p
        className="card-text text-secondary"
      >
        Precio total : test 1 precioTotal€
      </p>
      <a
        className="btn btn-outline-secondary border-0"
        onClick={[Function]}
        rel="noreferrer"
      >
        Cancelar
      </a>
      <a
        className="btn btn-outline-secondary border-0"
        href="test 1 urlDetalles"
        rel="noreferrer"
      >
        Detalles
      </a>
    </div>
  </div>
</div>
`;
